confidence.py

```diff
@@ -227,7 +227,8 @@ def readf(fname):
         if default is _NoDefault or path.exists(fname):
             # (attempt to) open fname if it exists OR if we're expected to raise an error on a missing file
             with open(fname, 'r') as fp:
-                return yaml.load(fp.read())
+                # default to empty dict, yaml.load will return None for an empty document
+                return yaml.load(fp.read()) or {}
```

Comment:
This fixes it! It does depend on ``yaml`` continuing to return ``None`` for empty sources in the future. This does not seem to be enforced by the tests in ``pyyaml``, but it is probably something that will not change. Furthermore I can't think of changes in ``yaml.load`` that would not break either ``test_loadf_empty``, or maybe even fix the same issue as is fixed here, so it's probably a safe assumption.

Reply:
Good call checking `pyyaml`, can't quite see if that's defined behaviour for them… Might be worth an issue on that end :)